### ISSUE TITLE
Unable to override default colors from extended Sass

### ISSUE THREAD

**Issue by msecret** [18F/web-design-standards#952]
In https://github.com/18F/cg-style, I have a scss file that has a file to override certain variables from the styleguide like color:

https://github.com/18F/cg-style/blob/master/src/css/override_vars.scss#L3
```
$color-primary:              #1F2B38;
$color-primary-darker:       #2C3E50;
$color-primary-darkest:      #1F2C38;
```

I then import the standards and include this file after:

https://github.com/18F/cg-style/blob/master/css/cg_style.scss#L4
```
@import "../node_modules/uswds/assets/_scss/all.scss";
@import "../src/css/override_vars.scss";
```

But when I use the CSS from *cg-style*, the colors of the USWDS components are not modified. For instance, the button color is still `#0071bc`. This includes if I import the overrides before or after the standards.

This is happening because the standards are setting all variables without `!default` in the `variables.scss` file. If this wasn't the case, and only the `defaults.scss` file was used, I would be able to import my override_vars.scss file and it would correctly be overriden.

**Comment by rossdakin** [comment]
@msecret this is great — I have been hoping for a way to override values without modifying the distributed flies (so I can just nuke and replace `/designstandards/` in my project but preserve my customized values).

Two potential issues with this though:
 1. I currently don't `@import "../node_modules/uswds/assets/_scss/all.scss";` (I just link to the precompiled CSS that you distribute) because the import breaks CSS paths to images. This is its own issue, not exacerbated by your change here but one that prevents me from benefitting from your change.
 2. Because I'm not importing `all`, I am importing `variables` so that I can at least reference them in my custom SCSS. This change will kill that file and thus break my code. I wonder if you could leave the file in place but just import `default` into it? Or just tell me "this is in beta -- get off my lawn." Both acceptable :-)

**Comment by maya** [comment]
+1 It would also be nice if we could provide `variables.scss` for people to download on the styleguide site to customize their variables and add the file themselves.

**Comment by rossdakin** [comment]
Also, just to make sure I understand this correctly, it seems like your example:

```scss
@import "../node_modules/uswds/assets/_scss/all.scss";
@import "../src/css/override_vars.scss";
```

…should actually be:

```scss
@import "../src/css/override_vars.scss";
@import "../node_modules/uswds/assets/_scss/all.scss";
```

Looking at the `!default` docs as referenced [here](https://robots.thoughtbot.com/sass-default) (but no longer live?):

> You can assign to variables if they aren’t already assigned by adding the !default flag to the end of the value. This means that if the variable has already been assigned to, it won’t be re-assigned, but if it doesn’t have a value yet, it will be given one.

So in the second version, you would set whatever override values you want, then `!default` would prevent them from being redefined in `default`, right?

**Comment by msecret** [comment]
@rossdakin you're correct on the last point.

We can keep the variables.scss file and not import it, that's fine.

I've been thinking about the file paths problem and I think we should explore dependency management soon. I'm interested in hearing more about how your build process works, do you have a link to the repo in question?

@maya, that also sounds fine as long as they're copying the file out of the uswds dir and inserting it into their repo. We should stop having people edit the libraries source files or we're setting them up for problems when they want to upgrade the standards.

**Comment by carodew** [comment]
Is this at all related to #960 ? Or two different things? (honest question)

**Comment by maya** [comment]
@carodew no, they are two separate issues.

**Comment by msecret** [comment]
This has been fixed in release 0.10.0

**Comment by rossdakin** [comment]
Awesome -- thanks all! 👏
